conn_linux.go

```diff
@@ -276,21 +277,67 @@ func (c *conn) SetWriteDeadline(t time.Time) error {
 // SetReadBuffer sets the size of the operating system's receive buffer
 // associated with the Conn.
 func (c *conn) SetReadBuffer(bytes int) error {
-	return os.NewSyscallError("setsockopt", c.s.SetSockoptInt(
+	// First try SO_RCVBUFFORCE. Given necessary permissions this syscall ignores limits.
+	err := os.NewSyscallError("setsockopt", c.s.SetSockoptInt(
 		unix.SOL_SOCKET,
-		unix.SO_RCVBUF,
+		unix.SO_RCVBUFFORCE,
 		bytes,
 	))
+	if err != nil {
+		// If SO_SNDBUFFORCE fails, try SO_RCVBUF
```

Comment:
Is there a specific syscall error we could inspect to determine if fallback is needed? Would something like `os.IsPermission` be sufficient?

Reply:
The man page does not require a specific error for that case, but the kernel uses `EPERM` (which is the only logical one of course). Also the permission check is the first thing to happen. I don't think it will ever make a difference if we check for permission errors or not, `libnfnetlink` does not check for it. Your choice.

Reply:
I'm good with leaving it be then, thanks for the info.